src/js/kernel.js

```diff
@@ -1182,27 +1182,74 @@ var kernel = execMain(function() {
 		return cc.ori || 0;
 	}
 
-	var longTouch = (function() {
-		var longTouchTid = 0;
+	var gestureRecognizer = (function () {
 
-		function startLongTouch() {
-			clearLongTouch();
-			longTouchTid = setTimeout(longTouchCallback, 2000);
+		var LONG_TOUCH_DURATION = 2000;
+		var SWIPE_THRESHOLD = 200;
```

Comment:
This threshold should be related to the screen size, but it’s not a big problem.

Reply:
Yep, maybe this should be investigated carefully, not sure but I though this is not just pixels but something like points depending on screen PPI. And 200 points seems reasonable default for most mobile screen sizes. Personally I've tested on 6.4-inch smartphone and 11-inch iPad.